card/lib/card/content.rb

```diff
@@ -134,85 +152,99 @@ def parse_content content
       ALLOWED_TAGS['table'] += %w[ cellpadding align border cellspacing ]
     end
 
-    ALLOWED_TAGS.each_key {|k|
+    ALLOWED_TAGS.each_key do |k|
       ALLOWED_TAGS[k] << 'class'
       ALLOWED_TAGS[k] << 'style' if Card.config.allow_inline_styles
       ALLOWED_TAGS[k]
-    }
+    end
     ALLOWED_TAGS
 
-    ATTR_VALUE_RE = [ /(?<=^')[^']+(?=')/, /(?<=^")[^"]+(?=")/, /\S+/ ]
+    ATTR_VALUE_RE = [/(?<=^')[^']+(?=')/, /(?<=^")[^"]+(?=")/, /\S+/]
 
     class << self
-
       ## Method that cleans the String of HTML tags
       ## and attributes outside of the allowed list.
 
       # this has been hacked for card to allow classes if
       # the class begins with "w-"
-      def clean!( string, tags = ALLOWED_TAGS )
-        string.gsub( /<(\/*)(\w+)([^>]*)>/ ) do
+      def clean!(string, tags=ALLOWED_TAGS)
+        string.gsub(/<(\/*)(\w+)([^>]*)>/) do
           raw = $~
           tag = raw[2].downcase
-          if attrs = tags[tag]
-            "<#{raw[1]}#{
+          if (attrs = tags[tag])
+            html_attribs =
               attrs.inject([tag]) do |pcs, attr|
-                q='"'
-                rest_value=nil
+                q = '"'
+                rest_value = nil
                 if raw[3] =~ /\b#{attr}\s*=\s*(?=(.))/i
                   rest_value = $'
-                  idx = %w{' "}.index($1) and q = $1
-                  re = ATTR_VALUE_RE[ idx || 2 ]
-                  if match = rest_value.match(re)
+                  (idx = %w{' "}.index($1)) && (q = $1)
+                  re = ATTR_VALUE_RE[idx || 2]
+                  if (match = rest_value.match(re))
                     rest_value = match[0]
                     if attr == 'class'
-                      rest_value = rest_value.split(/\s+/).find_all {|s| s=~/^w-/i}*' '
+                      rest_value =
+                        rest_value.split(/\s+/).select do |s|
+                          s =~ /^w-/i
+                        end * ' '
                     end
                   end
                 end
                 pcs << "#{attr}=#{q}#{rest_value}#{q}" unless rest_value.blank?
                 pcs
               end * ' '
-            }>"
+            "<#{raw[1]}#{html_attribs}>"
           else
-            " "
+            ' '
           end
         end.gsub(/<\!--.*?-->/, '')
       end
 
       if Card.config.space_last_in_multispace
-        def clean_with_space_last! string, tags = ALLOWED_TAGS
-          clean_without_space_last!(string, tags).gsub(/(?:^|\b) ((?:&nbsp;)+)/, '\1 ')
+        def clean_with_space_last! string, tags=ALLOWED_TAGS
+          cwo = clean_without_space_last!(string, tags)
+          cwo.gsub(/(?:^|\b) ((?:&nbsp;)+)/, '\1 ')
         end
         alias_method_chain :clean!, :space_last
       end
-      def truncatewords_with_closing_tags(input, words = 25, truncate_string = "...")
+
+      def truncatewords_with_closing_tags input, words=25, truncate_string='...'
```

Comment:
Unused method argument - `truncate_string`. If it's necessary, use `_` or `_truncate_string` as an argument name to indicate that it won't be used.<br>Assignment Branch Condition size for truncatewords_with_closing_tags is too high. [37.18/15]<br>Method has too many lines. [23/15]